### ISSUE TITLE
Configuration is lost sometimes

### ISSUE THREAD

**Issue by cburgmer** [build-canaries/nevergreen#99]
This is difficult to reproduce, but after upgrading to trunk, one of our build monitors keeps forgetting all the selected jobs. Even while reconfiguring this morning the config was lost twice out of three attempts.

The monitor in question possibly is restarted at night or due to automated updates. My expectations would be that the information stored in local storage should still be retained though.

Needless to say that with 100 jobs and only 20 selected this bug gets annoying pretty quickly :)

**Comment by GentlemanHal** [comment]
This is almost definitely because of this: https://build-canaries.github.io/2015/07/20/breaking-changes.html

Because we haven't release 0.7.0 yet the version is currently set to 0.7.0-alpha which is older than 0.7.0, so the code we added to fix the issue mentioned in the blog will keep clearing the local storage.

Check to see if you are getting a log message in the console

```
Clearing local storage as it was saved by an older version, see issue #94 for more details!
```

We plan to release 0.7.0 today so hopefully this will fix your issue!

**Comment by joejag** [comment]
Hi @cburgmer, thanks for reporting the issue. It probably is because of what @GentlemanHal was saying.

There's an outside chance your localStorage is being removed by another process, which would make Nevergreen not a great option for you.

Can you try going to another website and running this in the console:
```localStorage.bleh = 'saved'```. Next time Nevergreen config gets blasted away you can try seeing is that other webpage still has this data saved.

**Comment by GentlemanHal** [comment]
@cburgmer Master now contains the correct version so, if I'm correct, pulling will fix your issue.

**Comment by cburgmer** [comment]
Thanks, I'll try on Monday. 
@joejag The previous version was working fine. Can't tell the version, it's just the standalone.jar lying around.

Some feedback on the local storage if you like: We show the same config on two different monitors. We need to click through the same steps on two machines. It does end up in slightly different configs. I'm not sure if I'd prefer a config on the server side that I could put under version control as job names frequently change, but it does feel safer than local storage. Got similar feedback from another team.

**Comment by GentlemanHal** [comment]
@cburgmer Thanks for taking the time to give us valuable feedback, it's good to get the opinion of people outside the core team :)

Would Issue #64 help solve your configuration issue? You'd have to remember to export and import but that doesn't seem like much of an overhead.

**Comment by cburgmer** [comment]
For me this is still broken on the one machine (actual build monitor), as in the config gets dropped every night.
The other machine (pairing station) is unaffected.

I reverted back to the last stable version we were using, as our pipeline state is degrading right now with the team not noticing broken builds.

To me it looks like the "config syncing" or what ever part is responsible for dropping the config in the morning is a moving part that we don't fully understand.

**Comment by cburgmer** [comment]
I should add that our build monitor has 4 tabs open to cycle through automatically with 2 tabs showing the same nevergreen page.

**Comment by GentlemanHal** [comment]
There is no config syncing, the configuration is only saved in your local storage. 

Could you answer the following questions please:

1. Are you using [nevergreen.io](http://nevergreen.io) or running your own version locally?
2. If you are running locally:
  1. How are you running it locally? ([Docker Image](https://github.com/build-canaries/nevergreen-docker), [OS Package](https://github.com/build-canaries/nevergreen-ospackage), java -jar)
  2. What version does the UI show (bottom of the menu)?
  3. What version (if any) is saved into local storage?
  4. If you checked out the source, what git revision?
3. Did you see the message in the JS console I mentioned in my first reply?
4. Did you try @joejag suggestion to see if all local storage is getting dropped?
5. What stable version did you revert back to?
6. Do you reboot the build monitor machine every night?
7. What browser are you using?

I currently suspect you may still be running a 0.7.0-alpha version which has a known problem with removing config every time you reload (for the reason mentioned in my first reply).

**Comment by cburgmer** [comment]
I will try to find some time tomorrow and debug in an independent instance on the same machine. Debugging is made difficult, as I cannot take down the working version of the monitor.

I'll answer the questions once I have all the information.

**Comment by joejag** [comment]
Thanks @cburgmer, the two nevergreen tabs is interesting. 

Our model is you have one nevergreen config per browser. I'm assuming the same browser is showing nevergreen 2 times with the same config?

**Comment by cburgmer** [comment]
Yes, we cycle through some tabs, and as that would not show nevergreen often enough our devs thought that having more than just one tab would be the best solution to that.

1. own version
2. 
   i. java -jar
   ii. v0.8.0-alpha Heat Wave
   iii. n/a (cannot reproduce currently as I don't want to touch machine)
   iv. 7380a35fd6f5ed346ee8c340634145b05e04b443 + my patch in https://github.com/cburgmer/nevergreen/commit/cf76afe95a5234c1bca35ff12667b8eee27a8f4a
3. n/a (cannot reproduce currently as I don't want to touch machine)
4. n/a (cannot reproduce currently as I don't want to touch machine)
5. v0.5.0 Eggplant
6. Machine apparently switches off at night and comes back in the morning.
7. Chrome

**Comment by GentlemanHal** [comment]
Hmm, this is an interesting problem indeed! 

I'm unable to reproduce locally, on [staging.nevergreen.io](http://staging.nevergreen.io/) (which is 0.8.0-alpha) or [nevergreen.io](http://nevergreen.io/#/) (which is 0.7.0). 

I've never tried running multiple tabs (for an extended period), but I can't see why that would cause issues, especially if they were both just showing the monitor view.

I suspect @joejag might be onto something and *all* your local storage is getting cleared during the reboot, so I'd recommend giving his suggestion a try.

Without knowing what is saved into your local storage it's going to be very hard for us to investigate further.

**Comment by cburgmer** [comment]
I'll try to run another round next week. Thanks for your patience so far!

**Comment by GentlemanHal** [comment]
We just released v0.8.0 which made big changes to the internals of the UI and how it saves its configuration. So if you're able to upgrade and try again to see if the problem remains, that would be awesome. 

Until then, since none of the core team was able to reproduce the issue and the code has changed substantially I'm going to close this issue.

Feel free to reopen if you are able to provide more information.
